docs/dev/designs/RestFrontEnd.md

```diff
@@ -0,0 +1,136 @@
+The server will be a distinct frontend process called `siodb_rest_mgr`.
+There will be a new parameter to indicate the port number on which the process will
+listen: `rest_client.port = 443`. Default port 443`.
+
+The process will convert the REST queries into the IOMgr engine protobuf
+message (From REST to SQL).
+
+It would only work per table and allow users to push/retrieve data through REST.
+
+## Third-party libraries
+
+Siodb will use Oatpp (https://github.com/oatpp/oatpp) for the implementation of the REST server
+in the `siodb_rest_mgr` process.
+
+Also, we would leverage the JSON API from Protobuf as much as possible for conversion between
+message to JSON. E.g. when sending the IOMgr response to the client.
+
+## Authorization
+
+Siodb must provide a new table `sys_user_tokens` linked with `sys_users.trid`. In that table
+a columns `name` (text) and `value` (text) must store the token information for each user.
+
+It is then possible to generate a token with an SQL command:
+
+```sql
+ALTER USER <user_name> CREATE TOKEN '<token_name>';
+```
+
+It is possible to delete a token will this SQL commands:
+
+```sql
+ALTER USER <user_name> DROP TOKEN '<token_name>';
+```
+
+A user must provide the Siodb user name plus a valid token to access and manipulate the data:
+
+```curl --user "Siodb_user_name"  https://siodb-srv:80/<path>```
+
+## REST Paths (for the first version)
+
+### GET
+
+`/<database_name>/tables/<table_name>/rows`: Return a JSON of all data from the table <table_name>.
+`/<database_name>/tables/<table_name>/rows/<trid>`: Return a JSON of the data from
+the table <table_name> with TRID == <trid>.
+
+Example:
+
+```json
+...
+{
+   "col1":"ValCol1",
+   "col2":"ValCol2",
+   ...
+}
+```
+
+### POST
+
+`/databases/<database_name>/tables/<table_name>/rows`: Create a new row
+
+- Body: the name of the columns and their values: `{ "col1": "ValCol1", "col2": "ValCol2", ...}`
+- Update the row with TRID = <trid>
+- Return the Response protobuf converted into JSON.
+
+### PUT
+
+`/databases/<database_name>/tables/<table_name>/rows/<trid>`:
+
+- Body: the name of the columns and their values: `{ "col1": "ValCol1", "col2": "ValCol2", ...}`
+- Update the row with TRID = <trid>
+- Return the Response protobuf converted into JSON.
+
+### DELETE
+
+`/databases/<database_name>/tables/<table_name>/rows/<trid>`:
+
+- Body: none
+- Delete the row with TRID = <trid>
+- Return the Response protobuf converted into JSON.
+
+## Protocol
+
+### GET
+
+1. A client sends a query to the server:
+`curl --user "Siodb_user_name"  https://siodb-srv:80/<path>`
+2. `siodb_rest_mgr` received the request and query the IOMgr to see if the token is valid.
+    - If valid: it goes to step 3
+    - If not valid, it return the message `Status::CODE_401, "Unauthorized"`
+3. `siodb_rest_mgr` receive the request.
+4. `siodb_rest_mgr` convert the request into a `DatabaseEngineRequest` converting the query
```

Comment:
Read my commend below about PUT/POST/DELETE. Can't make any SQL in the rest server.